paint/PAINT_DrawWindow.cpp

```diff
@@ -30,48 +30,56 @@ DrawWindow::~DrawWindow()
 	}
 }
 
+/*override*/
+void DrawWindow::mouseButtonDown(MouseButton const b)
+{	
+	const win::Coords rel = { mouseCoords_.x - this->getRect().x, mouseCoords_.y - this->getRect().y };
+	const win::Coords prevRel = { prevMouseCoords_.x - this->getRect().x, prevMouseCoords_.y - this->getRect().y };
+	
```

Comment:
It's entirely possible that coordinates might be outside the rectangle of the draw window. You could reject them (ie don't draw) if they are, but a better mechanism is to perform _clipping_ which will cut the lines you are drawing off at the right place on the edges of the draw window.
**SHOULD**